### ISSUE TITLE
Should we keep duplicates in training data of pytorch-biggraph?

### ISSUE THREAD

**Issue by dbanka** [facebookresearch/PyTorch-BigGraph#181]
Should we keep duplicate values while training pytorch-biggraph graph embeddings model?

For example, If I have an entry value "A -> B" which came in training data 100 times. Should I keep it 100 times or take unique value and give it only once?

**Comment by lw** [comment]
That's really up to you and what you want your model to learn. Having multiple occurrences of an edge is in a sense equivalent to have that edge only once but with a higher "weight" (in fact, since we don't support proper weights, this is the only way to achieve such a behavior). Hence if in your use case a more frequent edge is indeed "more important" and should thus have more influence in the loss calculation, keep the duplicates. If not, remove them.

**Comment by dbanka** [comment]
thanks for the answer.. it cleared my doubt
